dotnet/samples/GettingStartedWithProcesses/SharedSteps/ScriptedUserInputStep.cs

```diff
@@ -48,22 +48,29 @@ public override ValueTask ActivateAsync(KernelProcessStepState<UserInputState> s
         return ValueTask.CompletedTask;
     }
 
+    internal string GetNextUserMessage()
+    {
+        var userMessage = this._state!.UserInputs[_state.CurrentInputIndex];
```

Comment:
Should there be any bounds check on this so we don't get an index out of bounds error? Thinking in the event that someone accidentally messes with the state elsewhere.

Reply:
added check + exit condition in the base class to emit exit event if the nextUserMessage is an empty string